# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserChild.h
# Commit: f8a2a169b296
# Full Hash: f8a2a169b2969d29e04733ae27a56459e5359973
# Author: Alexandru Marc <amarc@mozila.com>
# Date: 2025-05-16 09:19:19
# Regressor Bug: 1550462
# File Overlap Count: 1
# Description:
#   Revert "Bug 1550462 - part 3: Enable `pointerrawupdate` event r=smaug" for causing mochitest failures @ test_pointerrawupdate_event_count.html
#   
#   This reverts commit eb27445e0352d6b4df540adb40e21f72fc052e80.
#   
#   Revert "Bug 1550462 - part 2: Make `PresShell::HandleEvent` dispatch preceding `pointerrawupdate` event r=smaug,dom-core,edgar"
# ==============================================================================

diff -r 8d25773fc82f -r f8a2a169b296 dom/ipc/BrowserChild.h
--- a/dom/ipc/BrowserChild.h	Fri May 16 06:20:47 2025 +0000
+++ b/dom/ipc/BrowserChild.h	Fri May 16 09:30:09 2025 +0300
@@ -46,7 +46,6 @@
 class nsIRequest;
 class nsISerialEventTarget;
 class nsIWebProgress;
-class nsPIDOMWindowInner;
 class nsWebBrowser;
 class nsDocShellLoadState;
 
@@ -641,36 +640,14 @@
   // may reentry the event loop and access to the same hashtable. It's
   // called when dispatching some mouse events other than mousemove.
   void FlushAllCoalescedMouseData();
-
   void ProcessPendingCoalescedMouseDataAndDispatchEvents();
 
   void ProcessPendingCoalescedTouchData();
 
-  /**
-   * Dispatch an eMouseRawUpdate event for dispatching ePointerRawUpdate event
-   * into the DOM immediately when aPendingEvent will be dispatched later.
-   * This does nothing if there is no window which has at least one
-   * `pointerrawupdate` event listener.
-   */
-  void HandleMouseRawUpdateEvent(const WidgetMouseEvent& aPendingMouseEvent,
-                                 const ScrollableLayerGuid& aGuid,
-                                 const uint64_t& aInputBlockId);
-
   void HandleRealMouseButtonEvent(const WidgetMouseEvent& aEvent,
                                   const ScrollableLayerGuid& aGuid,
                                   const uint64_t& aInputBlockId);
 
-  /**
-   * Dispatch an eTouchRawUpdate event for dispatching ePointerRawUpdate event
-   * into the DOM immediately when aPendingEvent will be dispatched later.
-   * This does nothing if there is no window which has at least one
-   * `pointerrawupdate` event listener.
-   */
-  void HandleTouchRawUpdateEvent(const WidgetTouchEvent& aPendingTouchEvent,
-                                 const ScrollableLayerGuid& aGuid,
-                                 const uint64_t& aInputBlockId,
-                                 const nsEventStatus& aApzResponse);
-
   void SetCancelContentJSEpoch(int32_t aEpoch) {
     mCancelContentJSEpoch = aEpoch;
   }
@@ -735,13 +712,6 @@
   mozilla::ipc::IPCResult RecvDispatchToDropTargetAndResumeEndDragSession(
       bool aShouldDrop);
 
-  void OnPointerRawUpdateEventListenerAdded(const nsPIDOMWindowInner* aWindow);
-  void OnPointerRawUpdateEventListenerRemoved(
-      const nsPIDOMWindowInner* aWindow);
-  [[nodiscard]] bool HasPointerRawUpdateEventListeners() const {
-    return !!mPointerRawUpdateWindowCount;
-  }
-
  protected:
   virtual ~BrowserChild();
 
@@ -853,8 +823,6 @@
   Maybe<CodeNameIndex> mPreviousConsumedKeyDownCode;
   uint32_t mChromeFlags;
   uint32_t mMaxTouchPoints;
-  // The number of windows which may have ePointerRawUpdate event listener.
-  uint32_t mPointerRawUpdateWindowCount = 0;
   layers::LayersId mLayersId;
   CSSRect mUnscaledOuterRect;
   Maybe<bool> mLayersConnected;